Update docs on Change Secrets Provider details (#3849)
pulumi/docs: content/docs/intro/concepts/config.md

### Changing the Secrets Provider for a Stack

> As of Pulumi v2.8.0, it is possible to change the secrets provider for a stack as follows:

Author: I created the link as I know what it will be from my local machine

https://www.pulumi.com/docs/reference/cli/pulumi_stack_change_secrets_provider/

```bash
$ pulumi stack change-secrets-provider "<secrets-provider>"
```

This will change the encrypted secrets in the provider configuration and the stack's state file. The supported secrets providers are:

> * default (Pulumi KMS)

Reviewer: ```suggestion
* `default`
```

Also - probably should make all of these quoted since they are the formal names?

I was also going to suggest linking these to somewhere where we defined how to use each one - but couldn't find great docs on that.  We may want to either link to https://www.pulumi.com/docs/reference/cli/pulumi_stack_init/ or copy some of the content there to a topic in the main docs and then reference that from here?

Author: I added a link to the docs in the stack init page

* passphrase
* awskms
* azurekeyvault
* gcpkms
* hashivault

After the provider has been changed, you should be able to run `pulumi preview` and see no proposed changes.  Your configuration secrets

> and state files are now encrypted using the new secrets provider.  You can remove the `stack.json` and `config.json` files to ensure the plain text of the secrets is no longer available, as well as remove the temporary stack (`pulumi stack rm temp`).

Reviewer: ```suggestion
and state files are now encrypted using the new secrets provider.
```

### Changing the Secrets Provider for a Stack

> To change the secrets provider for an existing stack use the [`pulumi stack change-secrets-provider`](https://www.pulumi.com/docs/reference/cli/pulumi_stack_change-secrets-provider/) command.

Reviewer 2: ```suggestion
To change the secrets provider for an existing stack use the [`pulumi stack change-secrets-provider`]({{< relref "/docs/reference/cli/pulumi_stack_change-secrets-provider" >}}) command.
```

Author: Thanks!!

```bash
$ pulumi stack change-secrets-provider "<secrets-provider>"
```

This will change the encrypted secrets in the provider configuration and the stack's state file to use the new secrets provider.

> The [supported secrets providers](https://www.pulumi.com/docs/reference/cli/pulumi_stack_init/) are:

Reviewer 2: ```suggestion
The [supported secrets providers]({{< relref "/docs/reference/cli/pulumi_stack_init" >}}) are:
```

Author: Thanks!!
